apps/ejabberd/src/ejabberd_config.erl

```diff
@@ -422,30 +424,29 @@ replace_macros(Terms) ->
 %% @doc Split Terms into normal terms and macro definitions.
 -spec split_terms_macros(Terms :: [term()]) -> {[term()], [macro()]}.
 split_terms_macros(Terms) ->
-    lists:foldl(
-      fun(Term, {TOs, Ms}) ->
-              case Term of
-                  {define_macro, Key, Value} ->
-                      case is_atom(Key) and is_all_uppercase(Key) of
-                          true ->
-                              {TOs, Ms++[{Key, Value}]};
-                          false ->
-                              exit({macro_not_properly_defined, Term})
-                      end;
-                  Term ->
-                      {TOs ++ [Term], Ms}
-              end
-      end,
-      {[], []},
-      Terms).
+    lists:foldl(fun split_terms_macros_fold/2, {[], []}, Terms).
 
+-spec split_terms_macros_fold(any(), Acc) -> Acc when
+      Acc :: {[term()], [{Key :: any(), Value :: any()}]}.
+split_terms_macros_fold(Term, {TOs, Ms}) ->
+    case Term of
```

Comment:
It's more of a "Elvis happy" thing. Previously Elvis wasn't happy about it because there were 3 level of nasted blocks. I could also make exit_if_not_properly_defined function instead, if you think that, it would look better, then I'm gonna change it this way.